components/net_traits/bluetooth_blacklist.rs

```diff
@@ -0,0 +1,109 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use regex::Regex;
+use std::cell::RefCell;
+use std::collections::HashMap;
+use std::fs::File;
+use std::io::Read;
+use std::path::Path;
+use std::string::String;
+
+const SECURITY_ERROR: &'static str = "The operation is insecure.";
+const EMPTY_STRING: &'static str = "";
+const EXCLUDE_READS: &'static str = "exclude-reads";
+const EXCLUDE_WRITES: &'static str = "exclude-writes";
+const FILE_PATH: &'static str = "./resources/gatt_blacklist.txt";
+const VALID_UUID_REGEX: &'static str = "^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}";
+
+thread_local!(pub static BLUETOOTH_BLACKLIST: RefCell<Result<BluetoothBlacklist, String>> =
+                RefCell::new(BluetoothBlacklist::init()));
+
+pub struct BluetoothBlacklist(HashMap<String, ValueType>);
+
+#[derive(Eq, PartialEq)]
+pub enum ValueType {
+    Exclude,
+    ExcludeReads,
+    ExcludeWrites,
+}
+
+impl BluetoothBlacklist {
+
+    pub fn init() -> Result<BluetoothBlacklist, String> {
+        let mut btblist = BluetoothBlacklist(HashMap::new());
+        try!(btblist.fill_blacklist());
+        Ok(btblist)
+    }
+
+    pub fn is_excluded(&self, uuid: &str) -> Result<(), String> {
+        if self.0.get(uuid).map_or(false, |value| value.eq(&ValueType::Exclude)) {
+            return Err(String::from(SECURITY_ERROR));
+        }
+        Ok(())
+    }
+
+    pub fn is_excluded_from_reads(&self, uuid: &str) -> Result<(), String> {
+        try!(self.is_excluded(uuid));
+        if self.0.get(uuid).map_or(false, |value| value.eq(&ValueType::ExcludeReads)) {
+            return Err(String::from(SECURITY_ERROR));
+        }
+        Ok(())
+    }
+
+    pub fn is_excluded_from_writes(&self, uuid: &str) -> Result<(), String> {
+        try!(self.is_excluded(uuid));
+        if self.0.get(uuid).map_or(false, |value| value.eq(&ValueType::ExcludeWrites)) {
+            return Err(String::from(SECURITY_ERROR));
+        }
+        Ok(())
+    }
+
+    fn fill_blacklist(&mut self) -> Result<(), String> {
+        let path = Path::new(FILE_PATH);
+        let mut file = match File::open(&path) {
```

Comment:
Without it we get an error: "cannot borrow immutable local variable `file` as mutable" when we call the `read_to_string` method.